libraries/Microsoft.Bot.Builder.Dialogs/SkillDialog.cs

```diff
@@ -149,37 +173,31 @@ private static BeginSkillDialogOptions ValidateBeginDialogArgs(object options)
                 throw new ArgumentNullException(nameof(options), $"{nameof(dialogArgs.Activity)} is null in {nameof(options)}");
             }
 
-            // Only accept Message or Event activities
-            if (dialogArgs.Activity.Type != ActivityTypes.Message && dialogArgs.Activity.Type != ActivityTypes.Event)
-            {
-                // Just forward to the remote skill
-                throw new ArgumentException($"Only {ActivityTypes.Message} and {ActivityTypes.Event} activities are supported. Received activity of type {dialogArgs.Activity.Type}.", nameof(options));
-            }
+            ValidateBeginActivity(dc);
 
             return dialogArgs;
         }
 
         private async Task<Activity> SendToSkillAsync(ITurnContext context, Activity activity, string connectionName, CancellationToken cancellationToken)
         {
-            // Create a conversationId to interact with the skill and send the activity
-            var conversationIdFactoryOptions = new SkillConversationIdFactoryOptions
+            if (activity.Type == ActivityTypes.Invoke)
             {
-                FromBotOAuthScope = context.TurnState.Get<string>(BotAdapter.OAuthScopeKey),
-                FromBotId = DialogOptions.BotId,
-                Activity = activity,
-                BotFrameworkSkill = DialogOptions.Skill
-            };
-            var skillConversationId = await DialogOptions.ConversationIdFactory.CreateSkillConversationIdAsync(conversationIdFactoryOptions, cancellationToken).ConfigureAwait(false);
+                // Force ExpectReplies for invoke activities so we can get the replies right away and send them back to the channel if needed.
+                // This makes sure that the dialog will receive the Invoke response from the skill and any other activities sent, including EoC.
+                activity.DeliveryMode = DeliveryModes.ExpectReplies;
+            }
+
+            var skillConversationId = await CreateSkillConversationIdAsync(context, activity, cancellationToken).ConfigureAwait(false);
 
             // Always save state before forwarding
             // (the dialog stack won't get updated with the skillDialog and things won't work if you don't)
-            var skillInfo = DialogOptions.Skill;
             await DialogOptions.ConversationState.SaveChangesAsync(context, true, cancellationToken).ConfigureAwait(false);
 
+            var skillInfo = DialogOptions.Skill;
             var response = await DialogOptions.SkillClient.PostActivityAsync<ExpectedReplies>(DialogOptions.BotId, skillInfo.AppId, skillInfo.SkillEndpoint, DialogOptions.SkillHostEndpoint, skillConversationId, activity, cancellationToken).ConfigureAwait(false);
 
             // Inspect the skill response status
-            if (!(response.Status >= 200 && response.Status <= 299))
+            if (!response.IsSuccessStatusCode)
```

Comment:
>IsSuccessStatusCode [](start = 26, length = 19)

Again, This doesn't seem like it should on the payload.  This is a helper function ...